src/main/app/src/pages/ScheduleActivityPage.js

```diff
@@ -83,6 +45,55 @@ const ScheduleActivityPage = ({isLoggedIn, accessToken, userId}) => {
     updateGuest("");
   }
 
+  const randomFunction = (testData) => {
+
+    // arr that has all the indices at first, but then removes the indices thare taken into consideration.
+    const arr = new Array();
+
+    //populate the array elements as 0 to testData's length. 
+    for (let i = 0; i < testData.length; i++) {
+      arr.push(i);
+    }
+
+    // item1,2,3 are the random indices received
+    // from arr i have gotten index item1.
+    var item1 = arr[Math.floor(Math.random() * arr.length)];
+    arr.splice(item1, 1);
+
+    var item2 = arr[Math.floor(Math.random() * arr.length)];
+    arr.splice(item2, 1);
+
+    var item3 = arr[Math.floor(Math.random() * arr.length)];
+    arr.splice(item3, 1);
+
+    // add the elements from testData at these random indices and return them. 
+    const randomArray = new Array();
+    randomArray.push(testData[item1]);
+    randomArray.push(testData[item2]);
+    randomArray.push(testData[item3]);
```

Comment:
This process seems very repetitive. Did you consider using a list for this?

Reply:
So remember we asked David during our Friday's pod meeting? He said this need not have a loop since we need only three. I used an array cause the required functionality can be done with a javascript array. It has many useful functions!

Reply:
What I meant is not using a while loop and randomly select numbers until you have 3 (the old way you did it). The reason is there is no guarantee that your code will end.

Using a for loop is totally fine here. It allows you to avoid repeating the same logic 3 times. Also if you want to make more/less random items it is easy to change.